src/create-template/pvs.ts

```diff
@@ -120,21 +141,59 @@ export const processPvs = (
         createTemplateItem({
           pathId,
           partName: component.name,
-          partRevision: DefaultPartRevision,
+          partRevision: getRevisionId(component.vertexIndex, properties),
           fileName: component.shape_source.file_name,
           transform,
         })
       );
     }
-  };
+  }
 
-  const components = parse(fileData, {
-    attributeNamePrefix: '',
-    ignoreAttributes: false,
-  }).PV_FILE.section_structure.component;
-  if (verbose) console.log(`Found ${components.length} components.`);
+  recurse(components, rootComponent);
+  return items;
+}
+
+function createTemplateItem(
+  args: CreateTemplateItemArgs
+): ExtendedTemplateItem {
+  const suppliedId = args.pathId === '' ? '/' : args.pathId;
+  const parentId =
+    suppliedId === '/'
+      ? undefined
+      : suppliedId.split('/').slice(0, -1).join('/');
+  const t = args.transform;
+
+  return {
+    fileName: args.fileName,
+    name: args.partName,
+    parentId: parentId === '' ? '/' : parentId,
+    source: `/parts?filter[suppliedId]=${encodeURI(
+      args.partName
+    )}&include=part-revisions&filter[part-revisions][suppliedId]=${
+      args.partRevision
+    }`,
+    suppliedId,
+    transform: !t || is4x4Identity(t) ? undefined : toTransform(t),
+  };
+}
 
-  recurse(components, components[rootIndex(components, root)]);
+function getRevisionId(idx: number, properties?: Properties): string {
+  if (!properties) return DefaultPartRevision;
+
+  for (const sectionProp of properties.properties) {
+    const propCompRef = sectionProp.property_component_ref[idx];
```

Comment:
These use the same indexes as the `section_structure` part of the file. This is much nicer than the string compare I was doing to find the corresponding properties.